### PR TITLE
Updating the JIT to support Narrow, WidenLower, and WidenUpper for Vector512

### PR DESCRIPTION
**Pull request by tannergooding**
This makes progress towards https://github.com/dotnet/runtime/issues/80814

### CODE REVIEW

**Comment by tannergooding** [comment]
CC. @dotnet/jit-contrib 

This makes progress towards https://github.com/dotnet/runtime/issues/80814. This includes accelerating Vector128/256 with the newer AVX-512 instructions, allowing overall better throughput and pipelining.

For most `T` this is always a net improvement to codegen size and most often removes a memory access since we no longer require a constant. For narrowing `long`/`ulong` to `int`/`uint` this represents a 4 byte codegen size regression in the scenario nothing is coming from memory, but results in overall more consistent and faster code due to removal of an instruction, having a shorter dependency chain, and not requiring more expensive cross lane permutations.

File: src/coreclr/jit/lowerxarch.cpp [hunk 1]
```diff
@@ -5583,6 +5583,37 @@ void Lowering::ContainCheckStoreIndir(GenTreeStoreInd* node)
...
+                case NI_AVX512F_ConvertToVector256UInt32:
+                case NI_AVX512BW_ConvertToVector128Byte:
+                case NI_AVX512BW_ConvertToVector128SByte:
+                case NI_AVX512BW_ConvertToVector256Byte:
+                case NI_AVX512BW_ConvertToVector256SByte:
+                {
+                    // These intrinsics are "ins reg/mem, xmm"
+                    unsigned simdSize = hwintrinsic->GetSimdSize();
+
+                    if (simdSize != 16)
```
Comment (EgorBo) [inline src/coreclr/jit/lowerxarch.cpp:5602]: supposed to be `simdSize == 16` ?

Comment (tannergooding) [reply src/coreclr/jit/lowerxarch.cpp:5602]: Yep, thanks!
```suggestion
                    if (simdSize == 16)
```

**Review by BruceForstall — APPROVED** [review]

**Comment by tannergooding** [comment]
Resolved merge conflicts

**Comment by tannergooding** [comment]
SPMI replay failure is unrelated and has failed in a couple other runs: https://dev.azure.com/dnceng-public/public/_build?definitionId=150&_a=summary

CC. @kunalspathak as it looks related to the new VectorTableLookup support. It may be an AltJIT related issue.

**Comment by tannergooding** [comment]
Logged https://github.com/dotnet/runtime/issues/84536

**Pull request merged**